src/models/TalkMapper.php

```diff
@@ -355,6 +382,137 @@ public function save($data) {
     }
 
     /**
+     * Edit a talk.
+     *
+     * Accepts a subset of talk fields
+     *
+     * The data-array is expected to have the following keys:
+     *
+     * * talk_title
+     * * url_fiendly_talk_title
+     * * talk_description
+     * * slides_link
+     * * language (a value from the column lang:lang_name
+     * * start_date (a timestamp)
+     * * duration
+     * * speakers (an array of names)
+     * * category (a value from the column categories:title)
+     *
+     * @param array $talk    talk data to insert into the database.
+     * @param int   $talk_id The ID of the talk to be edited
+     *
+     * @return integer|false
+     */
+    public function edit($talk, $talk_id)
+    {
+        var_dump($talk);
+        // Sanity check: ensure all mandatory fields are present.
+        $mandatory_fields = array(
+            'talk_title',
+            'talk_description',
+            'start_date',
+            'duration',
+            'type',
+        );
+        $contains_mandatory_fields = !array_diff($mandatory_fields, array_keys($talk));
+        if (!$contains_mandatory_fields) {
+            throw new Exception("Missing mandatory fields");
+        }
+
+        $sql = "UPDATE talks SET %s WHERE ID = :talk_id";
+
+        // get the list of column to API field name for all valid fields
+        $fields = $this->getVerboseFields();
+        $items  = array();
+
+        foreach ($fields as $api_name => $column_name) {
+            // We don't change any activation stuff here!!
+            if (in_array($column_name, ['pending', 'active', 'average_rating', 'comments_enabled', 'comment_count', 'starred', 'starred_count', 'category'])) {
+                continue;
+            }
+            if ($column_name == 'lang_name') {
+                $pairs[] = "lang = (select ID from lang where lang_name = :$api_name)";
+                $items[$api_name] = $talk[$api_name];
+                continue;
+            }
+            if (isset($talk[$api_name])) {
+                $pairs[] = "$column_name = :$api_name";
+                $items[$api_name] = $talk[$api_name];
+            }
+        }
+
+        $items['talk_id'] = $talk_id;
+
+        $stmt = $this->_db->prepare(sprintf($sql, implode(', ', $pairs)));
+
+        if (! $stmt->execute($items)) {
+            throw new Exception(sprintf(
+                'executing "%s" resulted in an error: %s',
+                $stmt->queryString,
+                implode(' :: ', $stmt->errorInfo())
+            ));
+            return false;
+        }
+
+        if (isset($talk['category'])) {
```

Comment:
Should this be testing against `$data`? I should be able to add a category to talk that doesn't have any.